addons/cargo/functions/fnc_removeCargoItem.sqf

```diff
@@ -17,32 +17,33 @@
  */
 #include "script_component.hpp"
 
-params ["_itemClass", "_vehicle", ["_amount", 1]];
-TRACE_3("params",_itemClass,_vehicle,_amount);
+params ["_item", "_vehicle", ["_amount", 1]];
+TRACE_3("params",_item,_vehicle,_amount);
 
 private _loaded = _vehicle getVariable [QGVAR(loaded), []];
 
-private _itemsToRemove = [];
+private _itemsRemoved = 0;
 private _addedSpace = 0;
 
 {
-    if (count _itemsToRemove == _amount) exitWith {};
+    if (_itemsRemoved == _amount) exitWith {};
 
-    if (_x == _itemClass || {typeOf _x == _itemClass}) then {
-        _itemsToRemove pushBack _forEachIndex;
+    if (_x isEqualTo _item || {_x isEqualType objNull && {typeOf _x == _item}}) then {
+        INC(_itemsRemoved);
         ADD(_addedSpace,[_x] call FUNC(getSizeItem));
 
         if (_x isEqualType objNull) then {
             deleteVehicle _x;
         };
+        _loaded set [_forEachIndex, nil];
     };
 } forEach _loaded;
 
-{_loaded deleteAt _x} forEach _itemsToRemove;
+FILTER(_loaded,_x != nil);
```

Comment:
A quicker way to do this would be using `arrayIntersect` which removes all `nil` entries if used on the same array 👍